src/compile/legend.ts

```diff
@@ -106,6 +110,17 @@ export function values(legend: Legend) {
   return vals;
 }
 
+export function type(legend: Legend, fieldDef: FieldDef, channel: Channel) {
+  if (legend.type) {
+    return legend.type;
+  }
+
+  if (channel === COLOR && !fieldDef.bin && !fieldDef.timeUnit && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
```

Comment:
I'd say in general that default makes sense. `timeUnit` is similar to binning and so we should treat it as such. Users can override the default. I'm sure we can refine the default once we have vl in voyager. 

<img width="363" alt="screen shot 2016-12-11 at 10 44 14" src="https://cloud.githubusercontent.com/assets/589034/21082353/ca002fa2-bf8e-11e6-948d-7fe628e6191c.png">